src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -1208,26 +1221,51 @@ internal Pipe GetRedirectionPipe(ExecutionContext context, PipelineProcessor par
                 return new Pipe { NullPipe = true };
             }
 
-            CommandProcessorBase commandProcessor = context.CreateCommand("out-file", false);
-            Diagnostics.Assert(commandProcessor != null, "CreateCommand returned null");
-
-            // Previously, we mandated Unicode encoding here
-            // Now, We can take what ever has been set if PSDefaultParameterValues
-            // Unicode is still the default, but now may be overridden
+            // determine whether we're trying to set a variable by inspecting the file path
+            // if we can determine that it's a variable, we'll use Set-Variable rather than Out-File
```

Comment:
Just a note and not something you need to change, but it seems in the future, it would make more sense to have `Out-Provider` and have the provider subsystem determine which provider receives it rather than hardcoding to specific cases here

Reply:
I did think about that when I put this together and thought that the scenarios for redirection to something other than filesystem or variable were pretty limited, so I took this approach.

Reply:
I agree that a smaller change now makes sense.  Created https://github.com/PowerShell/PowerShell/issues/23794 to capture this.